dockershim removal readiness blog post (#30447)
kubernetes/website: content/en/blog/_posts/2021-11-12-are-you-ready-for-dockershim-removal/index.md

---
layout: blog
title: "Dockershim removal is coming. Are you ready?"
date: 2021-11-12
slug: are-you-ready-for-dockershim-removal
---

**Author:** Sergey Kanzhelev, Google. With reviews from Davanum Srinivas, Elana Hashman, Noah Kantrowitz, Rey Lejano.

> Last year we announced that Dockershim is being deprecated: [Dockershim Deprecation FAQ](https://kubernetes.io/blog/2020/12/02/dockershim-faq/).
> Our current plan is to remove the dockershim from Kubernetes codebase soon.

Reviewer: non-blocking nit: align case for dockershim
```suggestion
Last year we announced that dockershim is being deprecated: [Dockershim Deprecation FAQ](https://kubernetes.io/blog/2020/12/02/dockershim-faq/).
Our current plan is to remove the dockershim from Kubernetes codebase soon.
```

Reviewer 2: Non-blocking nit: Would it make sense here to link to the Google group notification that dims put out?

Reviewer 3: I think the blog post is a better option here - it supplies much more information than the group notification and has been translated into several other languages. 

Also seconding @reylejano's comment, I would convert it from a url to a path - that will make it default to the viewers preferred language.

We are looking for feedback from you whether you are ready for dockershim
removal and to ensure that you are ready when the time comes.
**Please fill out this survey: https://forms.gle/svCJmhvTv78jGdSx8**.

Dockershim component that enables Docker as a Kubernetes container runtime is

> being deprecated in favor of runtimes that directly use the [Container Runtime Interface](https://kubernetes.io/blog/2016/12/container-runtime-interface-cri-in-kubernetes/)

Reviewer: ```suggestion
being deprecated in favor of runtimes that directly use the [Container Runtime Interface](/blog/2016/12/container-runtime-interface-cri-in-kubernetes/)
```

created for Kubernetes. Many Kubernetes users have migrated to
other container runtimes without problems. However we see that dockershim is
still very popular. You may see some public numbers in recent [Container Report](https://www.datadoghq.com/container-report/#8) from DataDog.
Some Kubernetes hosting vendors just recently enabled other runtimes support
(especially for Windows nodes). And we know that many third party tools vendors

> are still not ready: [migrating telemetry and security agents](https://kubernetes.io/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents/#telemetry-and-security-agent-vendors).

Reviewer: ```suggestion
are still not ready: [migrating telemetry and security agents](/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents/#telemetry-and-security-agent-vendors).
```

At this point we believe that there is a feature parity between Docker and other
runtimes, many customers running production workload with other runtimes, and

> [migration documentation](https://kubernetes.io/docs/tasks/administer-cluster/migrating-from-dockershim/) is clear.

Reviewer: ```suggestion
[migration documentation](/docs/tasks/administer-cluster/migrating-from-dockershim/) is clear.
```

> Plan of record today is that dockershim will be removed in version 1.24.

Reviewer: non-blocking nit:
```suggestion
The plan of record today is that dockershim will be removed in version 1.24.
```

is an irreversible operation that will happen at the beginning of the 1.24
release cycle.

So there is only one month left to give us feedback. We want you to tell us how
ready you are.

**We are collecting opinions through this survey: [https://forms.gle/svCJmhvTv78jGdSx8](https://forms.gle/svCJmhvTv78jGdSx8)**
To better understand preparedness for the dockershim removal, our survey is
asking the version of Kubernetes you are currently using, and an estimate of
when you think you will adopt Kubernetes 1.24. All the aggregated information
on dockershim removal readiness will be published.
Free form comments will be reviewed by SIG Node leadership. If you want to
discuss any details of migrating from dockershim, report bugs or adoption
blockers, you can use one of the SIG Node contact options any time:
https://github.com/kubernetes/community/tree/master/sig-node#contact

Kubernetes is a mature enterprise ready product. This deprecation is another
step in the effort to get away from permanent beta features and providing more
stability and compatibility guarantees. With the migration from dockershim you
will get more flexibility and choice of container runtime features as well as

> less dependencies of your apps on specific underlying technology. Please take

Reviewer 4: To carry over my comment from the old doc, I an slightly worried about this last sentence as potentially reading as threatening but I am too brain fried to suggest a rewording. Anyone have suggestions?

Reviewer 2: +1 to Rey's suggestion. That sounds way less threatening

Author: I rephrased a bit including the sentence about consulting with k8s hosting vendor. Thank you!

Last year we announced that Dockershim is being deprecated: [Dockershim Deprecation FAQ](https://kubernetes.io/blog/2020/12/02/dockershim-faq/).
Our current plan is to remove the dockershim from Kubernetes codebase soon.
We are looking for feedback from you whether you are ready for dockershim
removal and to ensure that you are ready when the time comes.
**Please fill out this survey: https://forms.gle/svCJmhvTv78jGdSx8**.

Dockershim component that enables Docker as a Kubernetes container runtime is
being deprecated in favor of runtimes that directly use the [Container Runtime Interface](/blog/2016/12/container-runtime-interface-cri-in-kubernetes/)
created for Kubernetes. Many Kubernetes users have migrated to
other container runtimes without problems. However we see that dockershim is
still very popular. You may see some public numbers in recent [Container Report](https://www.datadoghq.com/container-report/#8) from DataDog.
Some Kubernetes hosting vendors just recently enabled other runtimes support
(especially for Windows nodes). And we know that many third party tools vendors

> are still not ready: [migrating telemetry and security agents](/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents/#telemetry-and-security-agent-vendors).

Author: Proposed merges hosting and tools vendors. I like mine better. Thank you for suggestion!

> At this point we believe that there is a feature parity between Docker and other
> runtimes, many customers running production workload with other runtimes, and
> [migration documentation](/docs/tasks/administer-cluster/migrating-from-dockershim/) is clear.
> The plan of record today is that dockershim will be removed in version 1.24.
> Development of this version will start in December and the removal of dockershim
> is an irreversible operation that will happen at the beginning of the 1.24
> release cycle.

Reviewer 3: ```suggestion
At this point, we believe that there is feature parity between Docker and the
other runtimes. Many end-users have used our [migration guide](/docs/tasks/administer-cluster/migrating-from-dockershim/)
and are running production workload using these different runtimes. The plan of
record today is that dockershim will be removed in version 1.24, slated for 
release around April of next year. For those developing or running alpha and
beta versions, dockershim will be removed in December at the beginning of the
1.24 release development cycle.
```

Reworded a bit to try and make it a little more clear on which audience will be impacted when with its removal.

> So there is only one month left to give us feedback. We want you to tell us how

Reviewer 3: ```suggestion
There is only one month left to give us feedback. We want you to tell us how
```

ready you are.

> **We are collecting opinions through this survey: [https://forms.gle/svCJmhvTv78jGdSx8](https://forms.gle/svCJmhvTv78jGdSx8)**

Reviewer 3: ```suggestion
**We are collecting opinions through this survey: https://forms.gle/svCJmhvTv78jGdSx8**

```

the markdown link around a link is redundant. Also added a blank line to give the survey a bit more emphasis

Author: markdown broke for me without the link syntax - it included the `**` into the link URL

To better understand preparedness for the dockershim removal, our survey is
asking the version of Kubernetes you are currently using, and an estimate of
when you think you will adopt Kubernetes 1.24. All the aggregated information
on dockershim removal readiness will be published.
Free form comments will be reviewed by SIG Node leadership. If you want to
discuss any details of migrating from dockershim, report bugs or adoption
blockers, you can use one of the SIG Node contact options any time:
https://github.com/kubernetes/community/tree/master/sig-node#contact

> Kubernetes is a mature enterprise ready product. This deprecation is another

Reviewer 3: ```suggestion
Kubernetes is a mature project. This deprecation is another
```
s/product/project - since this is an OSS blog post

step in the effort to get away from permanent beta features and providing more
stability and compatibility guarantees. With the migration from dockershim you
will get more flexibility and choice of container runtime features as well as
less dependencies of your apps on specific underlying technology. Please take
time to review the [dockershim migration documentation](/docs/tasks/administer-cluster/migrating-from-dockershim/)

> and consult your Kubernetes hosting vendor what container runtime options are available for you.

Reviewer 3: ```suggestion
and consult your Kubernetes hosting vendor (if you have one) what container runtime options are available for you.
```
